packages/preact/src/actions/updateSearch.ts

```diff
@@ -1,15 +1,23 @@
 import { Options as SearchConfig } from "@core/types"
 import type { SearchQuery } from "@nosto/nosto-js/client"
+import { SearchResultTransformer } from "@preact/hooks/useLoadMore/transformSearchResult"
 import { deepMerge } from "@utils/deepMerge"
 import { measure } from "@utils/performance"
 
 import { newSearch } from "./newSearch"
 import { ActionContext } from "./types"
 
-export async function updateSearch(context: ActionContext, query: SearchQuery, options?: SearchConfig): Promise<void> {
+export type UpdateSearchOptions = {
+  context: ActionContext
+  query: SearchQuery
+  options?: SearchConfig
+  transformer?: SearchResultTransformer
```

Comment:
are we expecting the need for multiple merge strategies?
if not, this could be something simpler like: `appendResults: boolean`

Reply:
This way it is more expandable also we could leave the logic for transforming result to the caller